### PR TITLE
Use DODataset to download a dataset from DO

### PR DESCRIPTION
**Pull request by dgaubert**
Please, I need an early review.

Still pending changes in `carto-python` to add `limit` and `order_by`.

### CODE REVIEW

File: cartoframes/data/observatory/catalog/entity.py [hunk 1]
```diff
@@ -126,35 +127,21 @@ def _download(self, credentials, file_path=None, limit=None, order_by=None):
         if not self._is_available_in('bq'):
             raise DOError('{} is not ready for Download. Please, contact us for more information.'.format(self))
 
-        bq_client = _get_bigquery_client(credentials)
```
Comment (rafatower) [inline cartoframes/data/observatory/catalog/entity.py]: You can probably do a cleanup of the unused stuff

Comment (dgaubert) [reply cartoframes/data/observatory/catalog/entity.py]: 👌

**Comment by dgaubert** [comment]
@rafatower, maybe we want to remove the class `BigQueryClient` as it seems unused from now on...

**Comment by dgaubert** [comment]
Ok, by removing unused method I need to make the ultimate sacrifice. :disappointed_relieved:

**Comment by rafatower** [comment]
Yeah, sure, if the class is not used anymore, delete it and related tests!

**Pull request merged**